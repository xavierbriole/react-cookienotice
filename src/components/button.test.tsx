import React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom'

import Button from './button'

describe('Button', () => {
  beforeEach(() => {
    jest.resetAllMocks()
  })

  it('should render', () => {
    const { container } = render(<Button label='label' onClick={() => {}} />)

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should display label', () => {
    const { getByText } = render(<Button label='label' />)

    expect(getByText('label')).toBeInTheDocument()
  })

  it('should call onClick', () => {
    const onClick = jest.fn()

    const { getByText } = render(<Button label='label' onClick={onClick} />)

    const button = getByText('label')

    button.click()

    expect(onClick).toHaveBeenCalled()
  })
})
