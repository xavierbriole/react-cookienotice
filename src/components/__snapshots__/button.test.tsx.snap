// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render 1`] = `
<button
  class="react-cookienotice-button"
>
  label
</button>
`;
